lib/jormungandr/src/Cardano/Wallet/Binary/Jormungandr.hs

```diff
@@ -106,21 +101,129 @@ getBlock = do
         $ whileM (not <$> isEmpty) getMessage
     return $ Block header msgs
 
+{-------------------------------------------------------------------------------
+                           Messages
+-------------------------------------------------------------------------------}
+
+-- | Messages are what the block body consists of.
+--
+-- Every message is prefixed with a message header.
+--
+--  Following, as closely as possible:
+-- https://github.com/input-output-hk/rust-cardano/blob/e0616f13bebd6b908320bddb1c1502dea0d3305a/chain-impl-mockchain/src/message/mod.rs#L22-L29
+data Message
+    = Initial [ConfigParam]
+    | OldUtxoDeclaration TODO
+    | Transaction TODO
+    | Certificate TODO
+    | UpdateProposal SignedUpdateProposal
+    | UpdateVote SignedVote
+    | UnimplementedMessage Int -- For development. Remove later.
+    deriving (Eq, Show)
+
 getMessage :: Get Message
 getMessage = do
     size <- fromIntegral <$> getWord16be
     contentType <- fromIntegral <$> getWord8
     let remaining = size - 1
     let unimpl = skip remaining >> return (UnimplementedMessage contentType)
     isolate remaining $ case contentType of
-        0 -> unimpl
+        0 -> Initial <$> getInitial
         1 -> unimpl
         2 -> unimpl
         3 -> unimpl
         4 -> unimpl
         5 -> unimpl
         other -> fail $ "Unexpected content type tag " ++ show other
 
+getInitial :: Get [ConfigParam]
+getInitial = do
+    len <- fromIntegral <$> getWord16be
+    replicateM len getConfigParam
+
+{-------------------------------------------------------------------------------
+                            Config Parameters
+-------------------------------------------------------------------------------}
+
+data ConfigParam
+    -- Seconds elapsed since 1-Jan-1970 (unix time)
+    = Block0Date Word64
+    | ConfigDiscrimination Discrimination
+    | ConsensusVersion Word16 -- ?
+    | SlotsPerEpoch Word32
+    | SlotDuration Word8
+    | EpochStabilityDepth Word32
+    | ConsensusGenesisPraosActiveSlotsCoeff Milli
+    | MaxNumberOfTransactionsPerBlock Word32
+    | BftSlotsRatio Milli
+    | AddBftLeader LeaderId
+    | RemoveBftLeader LeaderId
+    | AllowAccountCreation Bool
+    | ConfigLinearFee LinearFee
+    | ProposalExpiration Word32
+    deriving (Eq, Show)
+
+getConfigParam :: Get ConfigParam
+getConfigParam = do
+    -- The tag and the size/length of the config param is stored in a single
+    -- @Word16@.
+    --
+    -- 6 least-significant bits: length
+    -- 12 most-significant bits: tag
+    taglen <- getWord16be
+    let tag = taglen `shift` (-6)
+    let len = fromIntegral $ taglen .&. (63) -- 0b111111
```

Comment:
Had this bit-wise logic separate at first, but 
- `TagLen` would be completely uninteresting to the rest of the code
- Testing it indirectly through `getConfigParam . putConfigParam` roundtrip should be fine…? 🤔 or we will miss potential edge-cases?